### PR TITLE
feat: support STRM files in library as available content

### LINKED ISSUE #973: Support STRM files in library as available content
### Description

Noticed while doing a library sync that STRM files which have yet to be played were not shown as available. This seems to be because the mediaSource data is not complete yet until playback has taken place.

I've created a prototype which works for my needs: https://github.com/markjbear/jellyseerr/pull/1/files

### Desired Behavior

Presence of a STRM container file for an item in Emby / Jellyfin should result in the item being marked as available

### Additional Context

_No response_

### Code of Conduct

- [X] I agree to follow Jellyseerr's Code of Conduct

### PR DESCRIPTION
**Pull request by alanmilinovic**
#### Description
Presence of a STRM container file for an item in Emby / Jellyfin should result in the item being marked as available

#### To-Dos

- [X] Successful build `pnpm build`
- [ ] Translation keys `pnpm i18n:extract`
- [ ] Database migration (if required)

#### Issues Fixed or Closed

- Fixes #973

### CODE REVIEW

**Comment by alanmilinovic** [comment]
Can someone please review?

File: server/lib/scanners/jellyfin/index.ts [hunk 1]
```diff
@@ -96,7 +96,7 @@ class JellyfinScanner {
           (MediaStream) => MediaStream.Type === 'Video'
         ).some((MediaStream) => {
           return (MediaStream.Width ?? 0) <= 2000;
-        });
+        }) || (MediaSource.Container && MediaSource.Container === "strm");
```
Comment (gauthier-th) [inline server/lib/scanners/jellyfin/index.ts]: The first condition here is not necessary.

```suggestion
        }) || MediaSource.Container === 'strm';
```

**Review by gauthier-th — CHANGES_REQUESTED** [review]
Please fix all the whitespace changes you made.
You can format your code with the command `pnpm format`.

**Review by alanmilinovic — COMMENTED** [review]
removing trailing whitespace

**Review by alanmilinovic — COMMENTED** [review]
fixed whitespacing

File: server/lib/scanners/jellyfin/index.ts [hunk 2]
```diff
@@ -370,18 +370,18 @@ class JellyfinScanner {
             newSeasons.filter(
               (season) => season.status === MediaStatus.AVAILABLE
             ).length +
-              (media?.seasons.filter(
-                (season) => season.status === MediaStatus.AVAILABLE
-              ).length ?? 0) >=
+            (media?.seasons.filter(
+              (season) => season.status === MediaStatus.AVAILABLE
+            ).length ?? 0) >=
```
Comment (gauthier-th) [inline server/lib/scanners/jellyfin/index.ts:375]: Not fixed here

File: server/lib/scanners/jellyfin/index.ts [hunk 3]
```diff
@@ -370,18 +370,18 @@ class JellyfinScanner {
...
           const isAll4kSeasons =
             newSeasons.filter(
               (season) => season.status4k === MediaStatus.AVAILABLE
             ).length +
-              (media?.seasons.filter(
-                (season) => season.status4k === MediaStatus.AVAILABLE
-              ).length ?? 0) >=
+            (media?.seasons.filter(
+              (season) => season.status4k === MediaStatus.AVAILABLE
+            ).length ?? 0) >=
```
Comment (gauthier-th) [inline server/lib/scanners/jellyfin/index.ts:384]: Not fixed here

File: server/lib/scanners/jellyfin/index.ts [hunk 4]
```diff
@@ -404,8 +404,7 @@ class JellyfinScanner {
             // the lastSeasonChange field so we can trigger notifications
             if (newStandardSeasonAvailable > currentStandardSeasonAvailable) {
               this.log(
-                `Detected ${
-                  newStandardSeasonAvailable - currentStandardSeasonAvailable
+                `Detected ${newStandardSeasonAvailable - currentStandardSeasonAvailable
```
Comment (gauthier-th) [inline server/lib/scanners/jellyfin/index.ts:407]: Not fixed here

File: server/lib/scanners/jellyfin/index.ts [hunk 5]
```diff
@@ -415,8 +414,7 @@ class JellyfinScanner {
 
             if (new4kSeasonAvailable > current4kSeasonAvailable) {
               this.log(
-                `Detected ${
-                  new4kSeasonAvailable - current4kSeasonAvailable
+                `Detected ${new4kSeasonAvailable - current4kSeasonAvailable
```
Comment (gauthier-th) [inline server/lib/scanners/jellyfin/index.ts:417]: Not fixed here

File: server/lib/scanners/jellyfin/index.ts [hunk 6]
```diff
@@ -444,8 +442,8 @@ class JellyfinScanner {
               isAllStandardSeasons || shouldStayAvailable
                 ? MediaStatus.AVAILABLE
                 : media.seasons.some(
-                    (season) => season.status !== MediaStatus.UNKNOWN
-                  )
+                  (season) => season.status !== MediaStatus.UNKNOWN
+                )
```
Comment (gauthier-th) [inline server/lib/scanners/jellyfin/index.ts:446]: Not fixed here

File: server/lib/scanners/jellyfin/index.ts [hunk 7]
```diff
@@ -455,8 +453,8 @@ class JellyfinScanner {
                   media.seasons.some(
                     (season) => season.status4k !== MediaStatus.UNKNOWN
                   )
-                ? MediaStatus.PARTIALLY_AVAILABLE
-                : MediaStatus.UNKNOWN;
+                  ? MediaStatus.PARTIALLY_AVAILABLE
+                  : MediaStatus.UNKNOWN;
```
Comment (gauthier-th) [inline server/lib/scanners/jellyfin/index.ts]: Not fixed here

File: server/lib/scanners/jellyfin/index.ts [hunk 8]
```diff
@@ -472,8 +470,8 @@ class JellyfinScanner {
               status: isAllStandardSeasons
                 ? MediaStatus.AVAILABLE
                 : newSeasons.some(
-                    (season) => season.status !== MediaStatus.UNKNOWN
-                  )
+                  (season) => season.status !== MediaStatus.UNKNOWN
+                )
```
Comment (gauthier-th) [inline server/lib/scanners/jellyfin/index.ts:474]: Not fixed here

File: server/lib/scanners/jellyfin/index.ts [hunk 9]
```diff
@@ -495,8 +493,7 @@ class JellyfinScanner {
       }
     } catch (e) {
       this.log(
-        `Failed to process Jellyfin item. Id: ${
-          jellyfinitem.SeriesId ?? jellyfinitem.SeasonId ?? jellyfinitem.Id
+        `Failed to process Jellyfin item. Id: ${jellyfinitem.SeriesId ?? jellyfinitem.SeasonId ?? jellyfinitem.Id
```
Comment (gauthier-th) [inline server/lib/scanners/jellyfin/index.ts:496]: Not fixed here

**Review by gauthier-th — CHANGES_REQUESTED** [review]
Whitespaces are not fixed.

File: server/lib/scanners/jellyfin/index.ts [hunk 10]
```diff
@@ -96,7 +96,7 @@ class JellyfinScanner {
           (MediaStream) => MediaStream.Type === 'Video'
         ).some((MediaStream) => {
           return (MediaStream.Width ?? 0) <= 2000;
-        });
+        }) || MediaSource.Container === 'strm';
```
Comment (gauthier-th) [inline server/lib/scanners/jellyfin/index.ts:99]: By doing this, all STRM media will be considered non-4k media. Right now I don't know what method to use, but you have to find a way to retrieve the media definition.

**Review by gauthier-th — CHANGES_REQUESTED** [review]

**Comment by gauthier-th** [comment]
You have to find a way to retrieve the media definition of the STRM stream to be classified as non-4k or 4k. You can't classify every STRM as non-4k.

File: server/lib/scanners/jellyfin/index.ts (hunk 10 above)
Comment (alanmilinovic) [reply server/lib/scanners/jellyfin/index.ts:99]: hm, I also have no idea.

**Comment by gauthier-th** [comment]
Closing this as it has been solved by using a Jellyfin plugin: https://github.com/gauthier-th/JellyfinStrmExtract/

**Pull request closed without merging**